src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Helpers.cs

```diff
@@ -0,0 +1,138 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Pkcs;
+using System.Security.Cryptography.X509Certificates;
+using X509IssuerSerial = System.Security.Cryptography.Xml.X509IssuerSerial;
+
+namespace Internal.Cryptography
+{
+    internal static class Helpers
+    {
+        public static byte[] CloneByteArray(this byte[] a)
+        {
+            return (byte[])(a.Clone());
+        }
+
+        /// <summary>
+        /// This is not just a convenience wrapper for Array.Resize(). In DEBUG builds, it forces the array to move in memory even if no resize is needed. This should be used by
+        /// helper methods that do anything of the form "call a native api once to get the estimated size, call it again to get the data and return the data in a byte[] array."
+        /// Sometimes, that data consist of a native data structure containing pointers to other parts of the block. Using such a helper to retrieve such a block results in an intermittent
+        /// AV. By using this helper, you make that AV repro every time.
+        /// </summary>
+        public static byte[] Resize(this byte[] a, int size)
+        {
+            Array.Resize(ref a, size);
+#if DEBUG
+            a = a.CloneByteArray();
+#endif
+            return a;
+        }
+
+        public static CmsRecipientCollection DeepCopy(this CmsRecipientCollection recipients)
+        {
+            CmsRecipientCollection recipientsCopy = new CmsRecipientCollection();
+            foreach (CmsRecipient recipient in recipients)
+            {
+                X509Certificate2 originalCert = recipient.Certificate;
+                X509Certificate2 certCopy = new X509Certificate2(originalCert.Handle);
+                CmsRecipient recipientCopy = new CmsRecipient(recipient.RecipientIdentifierType, certCopy);
+                recipientsCopy.Add(recipientCopy);
+                GC.KeepAlive(originalCert);
+            }
+            return recipientsCopy;
+        }
+
+        public static X509Certificate2Collection GetStoreCertificates(StoreName storeName, StoreLocation storeLocation, bool openExistingOnly)
+        {
+            using (X509Store store = new X509Store())
+            {
+                OpenFlags flags = OpenFlags.ReadOnly | OpenFlags.IncludeArchived;
+                if (openExistingOnly)
+                    flags |= OpenFlags.OpenExistingOnly;
+
+                store.Open(flags);
+                X509Certificate2Collection certificates = store.Certificates;
+                return certificates;
+            }
+        }
+
+        public static X509Certificate2 TryFindMatchingCertificate(this X509Certificate2Collection certs, SubjectIdentifier recipientIdentifier)
```

Comment:
Hmm.. I don’t know that your assumption is universal – actually, I know it’s not because it’s not mine ☺

Let’s table this one as we’ll keep fighting this battle over and over... I’ve used this naming convention for a long time.